src/libgit2/refs.c

```diff
@@ -1054,10 +1054,14 @@ int git_reference_cmp(
 	const git_reference *ref2)
 {
 	git_reference_t type1, type2;
+	int ret;
 
 	GIT_ASSERT_ARG(ref1);
 	GIT_ASSERT_ARG(ref2);
 
+	if ((ret = strcmp(ref1->name, ref2->name)) != 0)
```

Comment:
Why is it necessary to continue inspecting the references if they have identical names? Is it so that "outdated" `git_reference` values compare as different?

Reply:
Right, that's the most likely scenario, I would expect. You load a reference, mutate it, and then check an existing reference to the new reference.